sled-agent/src/instance.rs

```diff
@@ -867,64 +985,228 @@ impl InstanceRunner {
                             port.name().into(),
                             port.slot().into(),
                         ),
+                    ))?;
+
+                let nic = &self.requested_nics[pos];
+                Ok(NicComponents {
+                    id: nic.id,
+                    device: VirtioNic {
+                        backend_id: SpecKey::Uuid(nic.id),
+                        interface_id: nic.id,
+                        // The old Propolis API added 8 to NIC slot numbers to
+                        // get their PCI device numbers.
+                        pci_path: PciPath::new(0, nic.slot + 8, 0)?,
+                    },
+                    backend: VirtioNetworkBackend {
+                        vnic_name: port.name().to_string(),
+                    },
+                })
+            })
+            .collect::<Result<Vec<NicComponents>, _>>()?;
+
+        // When a VM migrates, the migration target inherits most of its
+        // configuration directly from the migration source. The exceptions are
+        // cases where the target VM needs new parameters in order to interface
+        // with services on its sled or in the rest of the rack. These include
+        //
+        // - Crucible disks: the target needs to connect to its downstairs
+        //   instances with new generation numbers supplied from Nexus
+        // - OPTE ports: the target needs to bind its VNICs to the correct
+        //   devices for its new host, which devices may have different names
```

Comment:
```suggestion
        //   devices for its new host, which may have different names
```
extra "devices" i think? ("than" would reflow onto this line so if you agree then mind the rustfmt)